File: longest_common_prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {

        StringBuilder prefix = new StringBuilder();
        String first = strs[0];
        boolean flag = true;
        int count = 0;
        int i = 0;

        if(first.isEmpty()) {
            return first;
        }

        while(flag) {
            if(i >= first.length()) {
                break;
            }
            for(String s: strs) {
                if(s.length() == i) {
                    break;
                }
                if(first.charAt(i) == s.charAt(i)) {
                    count++;
                }
            }
            if(count != strs.length) {
                flag = false;
            } else {
                prefix.append(first.charAt(i));
            }
            count = 0;
            i++;
        }
        return prefix.toString();

    }
}

File: longest_common_prefix.py
# Write a function to find the longest common prefix string amongst an array of strings.
# 
# If there is no common prefix, return an empty string "".

class Solution(object):
    def longestCommonPrefix(self, strs):
        prefix = ""

        for i in range(len(strs[0])):
            for str in strs:
                if i == len(str) or str[i] != strs[0][i]:
                    return prefix
            prefix += strs[0][i]
        return prefix
        """
        :type strs: List[str]
        :rtype: str
        """


sol = Solution()

print(sol.longestCommonPrefix(["flower","flow","flight"]))
